tools/ddprobe.cpp

```diff
@@ -107,9 +229,24 @@ test_dxgi_duplication(dxgi::adapter_t &adapter, dxgi::output_t &output) {
   // Ensure we can duplicate the current display
   syncThreadDesktop();
 
-  // Return the result of DuplicateOutput() to Sunshine
+  // Attempt to duplicate the output
   dxgi::dup_t dup;
-  return output1->DuplicateOutput((IUnknown *) device.get(), &dup);
+  ComPtr<ID3D11Device> device_ptr(device.get());
+  HRESULT result = output1->DuplicateOutput(device_ptr.Get(), &dup);
+
+  if (FAILED(result)) {
+    std::cout << "Failed to duplicate output [0x"sv << util::hex(result).to_string_view() << "]" << std::endl;
+    return result;
+  }
+
+  // If duplication is successful, test frame capture
+  HRESULT captureResult = test_frame_capture(dup, device_ptr.Get());
```

Comment:
I'm concerned we may generate false negatives for previously working systems. Can we pass a parameter into ddprobe to allow us to do a first pass with your new check followed by a second pass without if the first one finds no matching configurations?

Reply:
Updated the code to make the frame capture point optional like you suggested